src/Core/Hook.php

```diff
@@ -0,0 +1,187 @@
+<?php
+/**
+ * @file src/Core/Hook.php
+ */
+namespace Friendica\Core;
+
+use Friendica\App;
+use Friendica\BaseObject;
+use Friendica\Database\DBA;
+
+/**
+ * Some functions to handle hooks
+ */
+class Hook extends BaseObject
+{
+	/**
+	 * Array of registered hooks
+	 *
+	 * Format:
+	 * [
+	 *		["<hook name>"] => [
+	 *			0 => "<hook file>",
+	 *			1 => "<hook function name>"
+	 *		],
+	 *		...
+	 * ]
+	 *
+	 * @var array
+	 */
+	private static $hooks = [];
+
+	/**
+	 * Load hooks
+	 */
+	public static function loadHooks()
+	{
+		self::$hooks = [];
+		$stmt = DBA::select('hook', ['hook', 'file', 'function'], [], ['order' => ['priority' => 'desc', 'file']]);
+
+		while ($hook = DBA::fetch($stmt)) {
+			if (!array_key_exists($hook['hook'], self::$hooks)) {
+				self::$hooks[$hook['hook']] = [];
+			}
+			self::$hooks[$hook['hook']][] = [$hook['file'], $hook['function']];
+		}
+		DBA::close($stmt);
+	}
+
+	/**
+	 * Registers a hook.
+	 *
+	 * @param string $hook     the name of the hook
+	 * @param string $file     the name of the file that hooks into
+	 * @param string $function the name of the function that the hook will call
+	 * @param int    $priority A priority (defaults to 0)
+	 * @return mixed|bool
+	 */
+	public static function register($hook, $file, $function, $priority = 0)
+	{
+		$file = str_replace(self::getApp()->getBasePath() . DIRECTORY_SEPARATOR, '', $file);
+
+		$condition = ['hook' => $hook, 'file' => $file, 'function' => $function];
+		if (DBA::exists('hook', $condition)) {
+			return true;
+		}
+
+		$result = DBA::insert('hook', ['hook' => $hook, 'file' => $file, 'function' => $function, 'priority' => $priority]);
+
+		return $result;
+	}
+
+	/**
+	 * Unregisters a hook.
+	 *
+	 * @param string $hook     the name of the hook
+	 * @param string $file     the name of the file that hooks into
+	 * @param string $function the name of the function that the hook called
+	 * @return boolean
+	 */
+	public static function unregister($hook, $file, $function)
+	{
+		$relative_file = str_replace(self::getApp()->getBasePath() . DIRECTORY_SEPARATOR, '', $file);
+
+		// This here is only needed for fixing a problem that existed on the develop branch
+		$condition = ['hook' => $hook, 'file' => $file, 'function' => $function];
+		DBA::delete('hook', $condition);
+
+		$condition = ['hook' => $hook, 'file' => $relative_file, 'function' => $function];
+		$result = DBA::delete('hook', $condition);
+		return $result;
+	}
+
+	/**
+	 * Returns the list of callbacks for a single hook
+	 *
+	 * @param  string $name
```

Comment:
`@param string $name Name of the hook`

Reply:
I'm adding it because I'm otherwise modifying the file, but this is completely unwarranted given the name the class, method, parameter and the description of the method.